packages/react-components/react-table/Spec.md

```diff
@@ -2,62 +2,373 @@
 
-- _Component States_
-- _Interaction_
-  - _Keyboard_
-  - _Cursor_
-  - _Touch_
-  - _Screen readers_
+This navigation mode can cause screen reader issues since tables are not intended to be navigated by row in any mode.
+This mode only navigates the table by row and can be useful when row selection is the only interactive feature of
+the component
+
+#### composite
+
+This navigation mode can cause screen reader issues since it is a combination of `row` and `cell` navigation modes.
+There is also no real standardized pattern for any 2 dimenstional navigation that supports both rows and cells. However
+to support existing legacy and product needs, it will be supported in the Table component.
+
+### Nested focusables in cells
+
+#### Single focusable
+
+When there is a single focusable element inside a cell, users are recommended to choose `cell` navigation mode. However
+`composite` navigation mode is also possible. In this scenario, cells will be focused on navigation, but the focusable
+element inside the cell should be focused if it exists.
+
+#### Nested focusable
+
+When there are multile focusable elemnts inside a cell, we implement a pattern similar to the [WAI grid pattern](https://www.w3.org/WAI/ARIA/apg/patterns/grid/).
+Pressing `Enter` on a cell will move focus and trap focus inside until the user presses `Escape` to revert back to grid navigation.
 
 ## Accessibility
 
-Base accessibility information is included in the design document. After the spec is filled and review, outcomes from it need to be communicated to design and incorporated in the design document.
-
-- Decide whether to use **native element** or folow **ARIA** and provide reasons
-- Identify the **[ARIA](https://www.w3.org/TR/wai-aria-practices-1.2/) pattern** and, if the component is listed there, follow its specification as possible.
-- Identify accessibility **variants**, the `role` ([ARIA roles](https://www.w3.org/TR/wai-aria-1.1/#role_definitions)) of the component, its `slots` and `aria-*` props.
-- Describe the **keyboard navigation**: Tab Oder and Arrow Key Navigation. Describe any other keyboard **shortcuts** used
-- Specify texts for **state change announcements** - [ARIA live regions
-  ](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions) (number of available items in dropdown, error messages, confirmations, ...)
-- Identify UI parts that appear on **hover or focus** and specify keyboard and screen reader interaction with them
-- List cases when **focus** needs to be **trapped** in sections of the UI (for dialogs and popups or for hierarchical navigation)
-- List cases when **focus** needs to be **moved programatically** (if parts of the UI are appearing/disappearing or other cases)
+The spec aims to use the accessibility section as little as possible and building an accessible component by default.
+The follow a11y resources were used in the drafting of this spec:
+
+- https://www.w3.org/WAI/ARIA/apg/patterns/grid/
```

Comment:
After reading this ARIA pattern it sounds to me like this spec is describing a component called `Grid` rather than `Table` ([the ARIA table spec for comparison](https://www.w3.org/WAI/ARIA/apg/patterns/table/)).

One thought I have is that `Table` and `Grid` are somewhat like `Combobox` and `Dropdown` in that they are similar controls and may have cases where they can be used interchangeably but also serve distinct purposes.

It also seems to me that controls like v8's DetailsList gets overloaded because we don't offer a rich set of list and grid-like controls in v8 and that splitting up functionality is a good way to keep individual component complexity down. For example, [ARIA table doesn't have any keyboard interactions](https://www.w3.org/WAI/ARIA/apg/patterns/table/#keyboard-interaction-20) which would make its implementation much simpler than a grid.

What do you think about renaming this component to `Grid` and having a separate `Table` component?

Reply:
`DataGrid` is definitely a better name :)

I'm in favor of keeping the static and interactive components separate. Doing so will keep the implementations of `Table` and `DataGrid` simpler and more focused and users that only need a static table don't have to pay the costs of all the interactivity of a `DataGrid`